### PR TITLE
Avro import for feed creation schema discovery

### PR DESCRIPTION
**Pull request by calvin-pietersen**
Added arvo imports to support avro sample file schema discovery during feed creation. Need to add dependancy jar "com.databricks.spark.avro" to spark class path to use Avro extensions.

https://mvnrepository.com/artifact/com.databricks/spark-avro_2.10
https://kylo-io.atlassian.net/browse/KYLO-407

### CODE REVIEW

File: plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java [hunk 1]
```diff
@@ -92,12 +92,11 @@ public Schema doParse(InputStream inputStream, SparkFileType fileType, TableSche
             return toSchema(response.getResults(), fileType, tableSchemaType);
 
         } catch (Exception e) {
-            log.warn("Error parsing file {}", fileType);
+            log.error("Error parsing file {}: {}", fileType, e.getMessage());
             throw new IOException("Unexpected exception. Verify file is the proper format", e);
         } finally {
             tempFile.delete();
```
Comment (KyloTech) [inline plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java]: MINOR Do something with the "boolean" value returned by "delete". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS899)

**Comment by KyloTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SparkFileSchemaParserService.java#L9](https://github.com/Teradata/kylo/blob/b3eb683ecac545b32c39d75f3cc896f9ab178689/plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [SparkFileSchemaParserService.java#L112](https://github.com/Teradata/kylo/blob/b3eb683ecac545b32c39d75f3cc896f9ab178689/plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java#L112): This block of commented-out lines of code should be removed. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [SparkFileSchemaParserService.java#L114](https://github.com/Teradata/kylo/blob/b3eb683ecac545b32c39d75f3cc896f9ab178689/plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java#L114): Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1149)
1. ![MINOR][MINOR] [SparkFileSchemaParserService.java#L143](https://github.com/Teradata/kylo/blob/b3eb683ecac545b32c39d75f3cc896f9ab178689/plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java#L143): Replace this "switch" statement by "if" statements to increase readability. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java (hunk 1 above)
Comment (freastro) [inline plugins/schema-discovery-default/src/main/java/com/thinkbiganalytics/discovery/parsers/hadoop/SparkFileSchemaParserService.java]: Please ignore.

**Comment by freastro** [comment]
The SonarQube results seem unrelated to your changes. Please ignore.

**Pull request merged**